mergify_engine/subscription.py

```diff
@@ -115,22 +116,45 @@ def to_dict(self) -> SubscriptionDict:
             "features": [f.value for f in self.features],
         }
 
+    RETENTION_SECONDS = 60 * 60 * 24 * 3  # 3 days
+    VALIDITY_SECONDS = 3600
+
+    async def _have_expired(self) -> bool:
+        expire_at: int = await self.redis.ttl(
+            f"subscription-cache-owner-{self.owner_id}"
+        )
+        if expire_at < 0:
+            return True
+
+        return expire_at < (self.RETENTION_SECONDS - self.VALIDITY_SECONDS)
+
     @classmethod
     async def get_subscription(
         cls, redis: utils.RedisCache, owner_id: int
     ) -> "Subscription":
         """Get a subscription."""
-        sub = await cls._retrieve_subscription_from_cache(redis, owner_id)
-        if sub is None:
-            sub = await cls._retrieve_subscription_from_db(redis, owner_id)
-            await sub.save_subscription_to_cache()
-        return sub
+
+        cached_sub = await cls._retrieve_subscription_from_cache(redis, owner_id)
```

Comment:
might be better to return the ttl at the same time using a transaction

Reply:
done